src/activities/CreateMeasurement2D.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import type {
+    IActivityHandler,
+    IActivityContext,
+} from "@geocortex/workflow/runtime/IActivityHandler";
+import { MapProvider } from "@geocortex/workflow/runtime/activities/arcgis/MapProvider";
+import { activate } from "@geocortex/workflow/runtime/Hooks";
+import DistanceMeasurement2D from "@arcgis/core/widgets/DistanceMeasurement2D";
+import AreaMeasurement2D from "@arcgis/core/widgets/AreaMeasurement2D";
+import MapView from "@arcgis/core/views/MapView";
+
+export interface CreateMeasurement2DInputs {
+
+    /**
+     * @description The type of measurement.
+     * @required
+     */
+    measurementType?: "area" | "distance" | string;
```

Comment:
We should consider making four activities. Anytime you have an input that sets a "mode" for the activity and then uses some of the inputs and outputs it can be trouble. 

There would be more boilerplate to create four activities, but they would each be dirt simple with essentially no conditional logic in them, so they would be easier to maintain. They would also be easier to extend because the inputs/outputs only have to concern what is supported by one widget/view-model.

Reply:
Done